certz/certz.proto

```diff
@@ -636,8 +641,84 @@ message CertificateSigningRequest {
 // the specified parameters.
 message GenerateCSRRequest {
   CSRParams params = 1;
+
+  // Indicates measurements that are prerequisites for certificate
+  // provisioning.
+  repeated ReferenceIntegritySpec integrity = 2;
 }
+
 // A message sent by the target as response to the GenerateCSRRequest.
 message GenerateCSRResponse {
   CertificateSigningRequest certificate_signing_request = 1;
+
+  // Reference integrity measurements as specified in the request.
+  repeated ReferenceIntegrityData integrity = 2;
+}
+
+message GetIntegrityManifestRequest {}
+
+message GetIntegrityManifestResponse {
+  bytes manifest = 1;
+}
+
+// See the TCG Reference Integrity Manifest (RIM) Information Model.
+message ReferenceIntegritySpec {
+  // Identifies the Root of Trust component.
+  string rot_id = 1;
+
+  oneof type {
+    MBMSpec mbm = 3;
+  }
+}
+
+// See the TCG Reference Integrity Manifest (RIM) Information Model.
+message ReferenceIntegrityData {
+  // Identifies the Root of Trust component.
+  string rot_id = 1;
+
+  // Contains reference measurements for the RoT component.
+  oneof type {
+    MBMData mbm = 2;
+  }
+}
+
+// Measured Boot Manifest specification.
+// Fields of type bytes are octets; not text, not base64 encoded.
+message MBMSpec {
+  // Template for an ephemeral signing key.
+  // This is a packed TCG data structure.
+  bytes ak_template = 1;
+
+  // Random nonce used to prevent replay attacks.
+  bytes ak_nonce = 2;
+
+  // Measurement selection structure.
+  // This is a packed TCG data structure.
+  bytes selection = 3;
+
+  // Random nonce used to prevent replay attacks.
+  bytes nonce = 4;
+}
+
+// Measured Boot Manifest measurements.
+// Fields of type bytes are octets; not text, not base64 encoded.
```

Comment:
can you provide a pointer to those structs? and when you say "octets" are you meaning a byte representation of a c struct?

Reply:
Yes, I mean a byte representation of a TSS struct. I've added the relevant links and struct names in the docstrings.